code/game/objects/effects/mines.dm

```diff
@@ -14,14 +14,12 @@
 /obj/effect/mine/Crossed(AM as mob|obj)
 	if(!isliving(AM))
 		return
-	if(isanimal(AM))
-		var/mob/living/simple_animal/SA = AM
-		if(faction && (faction in SA.faction))
-			return
-		if(!SA.flying)
-			triggermine(SA)
-	else
-		triggermine(AM)
+	var/mob/living/M = AM

```

Comment:
Aside from being more cleanly written, this change also means that players who visit the depot in peaceful mode will no longer trigger depot mines, which is consistent with how other depot defenses work.